### PR TITLE
Fix static check failures for pkg/controller/podautoscaler

### PR DESCRIPTION
**Pull request by sadlil**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Fixes the following static check failures for pkg/controller/podautoscaler:
```
pkg/controller/podautoscaler/horizontal_test.go:2481:6: func NoTestBackoffUpscale is unused (U1000)
pkg/controller/podautoscaler/legacy_horizontal_test.go:756:6: func LegacyTestTolerance is unused (U1000)
pkg/controller/podautoscaler/legacy_horizontal_test.go:770:6: func LegacyTestToleranceCM is unused (U1000)
pkg/controller/podautoscaler/legacy_horizontal_test.go:791:6: func LegacyTestMinReplicas is unused (U1000)
pkg/controller/podautoscaler/legacy_horizontal_test.go:805:6: func LegacyTestZeroReplicas is unused (U1000)
pkg/controller/podautoscaler/legacy_horizontal_test.go:819:6: func LegacyTestTooFewReplicas is unused (U1000)
pkg/controller/podautoscaler/legacy_horizontal_test.go:833:6: func LegacyTestTooManyReplicas is unused (U1000)
pkg/controller/podautoscaler/legacy_horizontal_test.go:847:6: func LegacyTestMaxReplicas is unused (U1000)
pkg/controller/podautoscaler/legacy_horizontal_test.go:875:6: func LegacyTestMissingMetrics is unused (U1000)
pkg/controller/podautoscaler/legacy_horizontal_test.go:889:6: func LegacyTestEmptyMetrics is unused (U1000)
pkg/controller/podautoscaler/legacy_horizontal_test.go:903:6: func LegacyTestEmptyCPURequest is unused (U1000)
pkg/controller/podautoscaler/legacy_horizontal_test.go:916:6: func LegacyTestEventCreated is unused (U1000)
pkg/controller/podautoscaler/legacy_horizontal_test.go:931:6: func LegacyTestEventNotCreated is unused (U1000)
pkg/controller/podautoscaler/legacy_horizontal_test.go:946:6: func LegacyTestMissingReports is unused (U1000)
pkg/controller/podautoscaler/legacy_horizontal_test.go:960:6: func LegacyTestUpscaleCap is unused (U1000)
pkg/controller/podautoscaler/legacy_horizontal_test.go:977:6: func LegacyTestComputedToleranceAlgImplementation is unused (U1000)
pkg/controller/podautoscaler/replica_calculator_test.go:94:2: field metricLabelSelector is unused (U1000)
```
**Which issue(s) this PR fixes**:

Fixes #
Ref #81657
**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/controller/podautoscaler/legacy_horizontal_test.go [hunk 1]
```diff
@@ -753,111 +752,6 @@ func TestLegacyScaleDownIgnoresUnreadyPods(t *testing.T) {
 	tc.runTest(t)
 }
 
-func LegacyTestTolerance(t *testing.T) {
```
Comment (liggitt) [inline pkg/controller/podautoscaler/legacy_horizontal_test.go]: @kubernetes/sig-autoscaling-misc should these be removed or promoted back to active tests?

**Comment by SataQiu** [comment]
/approve
for hack

File: pkg/controller/podautoscaler/legacy_horizontal_test.go (hunk 1 above)
Comment (sadlil) [reply pkg/controller/podautoscaler/legacy_horizontal_test.go]: Okay. Let me try to promote them back to active tests. Thanks for your feedback. :)

**Comment by sadlil** [comment]
I will create a new pr trying to upgrade the tests to active test.

**Pull request closed without merging**